querylog/file_writer_test.go

```diff
@@ -117,21 +115,11 @@ var _ = Describe("FileWriter", func() {
 					})
 				})
 
-				Eventually(func(g Gomega) int {
-					filesCount, err := tmpDir.CountFiles()
-					g.Expect(err).Should(Succeed())
-
-					return filesCount
-				}, "20s", "1s").Should(Equal(2))
+				Expect(tmpDir.ReadDir()).Should(HaveLen(2))
```

Comment:
I think that will potentially increase flakiness since in theory your approach is correct but will fail on filesystem lags because the directory lookup is performed directly after the second write.

If I remember correctly the `Eventually` was added because it was flaky on the GitHub runner. 🤔

Reply:
Yeah `git blame` confirms that, I thought is was because of doing the cleanup in another goroutine.  
I reverted to `Eventually` and decided to remove `TempFolder.ReadDir()` since this was the only use.